diff --git a/utils/useScrollLock.js b/utils/useScrollLock.js
index 58ccee9f790bd2d64f04a642c115f36fc73e9264..eff9fe5b4185304481be193d60b3a56ee9b3485a 100644
--- a/utils/useScrollLock.js
+++ b/utils/useScrollLock.js
@@ -124,7 +124,9 @@ function preventScrollStandard(referenceElement) {
   return () => {
     resizeFrame.cancel();
     cleanup();
-    win.removeEventListener('resize', handleResize);
+    if (win.removeEventListener) {
+      win.removeEventListener('resize', handleResize);
+    }
   };
 }
 class ScrollLocker {
