diff --git a/utils/useScrollLock.js b/utils/useScrollLock.js
index 58ccee9f790bd2d64f04a642c115f36fc73e9264..14c95eee35d54ecae013f1f4f07ff29e66f4bb1f 100644
--- a/utils/useScrollLock.js
+++ b/utils/useScrollLock.js
@@ -124,7 +124,9 @@ function preventScrollStandard(referenceElement) {
   return () => {
     resizeFrame.cancel();
     cleanup();
-    win.removeEventListener('resize', handleResize);
+    if (typeof win.removeEventListener === 'function') {
+      win.removeEventListener('resize', handleResize);
+    }
   };
 }
 class ScrollLocker {
