diff --git a/esm/useLazyRipple/useLazyRipple.js b/esm/useLazyRipple/useLazyRipple.js
index 1c6f64f7b2680bc72a8766653fd6a837443b2cf9..bc4509f4891adbbce3480d4cfe9dc0d50bfaac5d 100644
--- a/esm/useLazyRipple/useLazyRipple.js
+++ b/esm/useLazyRipple/useLazyRipple.js
@@ -55,6 +55,9 @@ export class LazyRipple {
         this.mounted.resolve();
       }
     }
+    return () => {
+      this.ref.current = null;
+    };
   };
 
   /* Ripple API */
diff --git a/useLazyRipple/useLazyRipple.js b/useLazyRipple/useLazyRipple.js
index 4120dd53afe4cfe96a7483a9cb5a920e6fc10377..50550332ac83237171652ccdd2cda73ce7bce90f 100644
--- a/useLazyRipple/useLazyRipple.js
+++ b/useLazyRipple/useLazyRipple.js
@@ -63,6 +63,9 @@ class LazyRipple {
         this.mounted.resolve();
       }
     }
+    return () => {
+      this.ref.current = null;
+    };
   };
 
   /* Ripple API */
