diff --git a/prepareMarkdown.mjs b/prepareMarkdown.mjs
index b188e747e54ba299ec857ec0d7c96bf39bce46e9..bd738deaa6a0721dc0588f0f99cb6ed2fd5ae8bd 100644
--- a/prepareMarkdown.mjs
+++ b/prepareMarkdown.mjs
@@ -41,6 +41,9 @@ function resolveComponentApiUrl(productId, componentPkg, component) {
   if (productId === 'x-data-grid') {
     return `/x/api/data-grid/${kebabCase(component)}/`;
   }
+  if (productId === 'x-chat') {
+    return `/x/api/chat/${kebabCase(component)}/`;
+  }
   if (componentPkg === 'mui-base' || BaseUIReexportedComponents.includes(component)) {
     return `/base-ui/react-${kebabCase(component)}/components-api/#${kebabCase(component)}`;
   }
