diff --git a/loader.mjs b/loader.mjs
index 5a68a3fb78e23ddb98a87750acd0ecd94be77171..1f979e818da87ad9681a0b43ef74940cefec5584 100644
--- a/loader.mjs
+++ b/loader.mjs
@@ -660,7 +660,7 @@ export default async function demoLoader() {
   );

   componentNames.forEach((componentName) => {
-    const moduleID = componentName.startsWith('@mui/docs/')
+    const moduleID = componentName.startsWith('@mui/internal-core-docs/')
       ? componentName
       : path.join(this.rootContext, 'src', componentName).replace(/\\/g, '/');

diff --git a/prepareMarkdown.mjs b/prepareMarkdown.mjs
index 2b2118bac260a00b8b33681a8a1c9dfda10dccd7..f5e34251b84db61848498f186cf8d2d7d1c565f0 100644
--- a/prepareMarkdown.mjs
+++ b/prepareMarkdown.mjs
@@ -44,6 +44,9 @@ function resolveComponentApiUrl(productId, componentPkg, component) {
   if (productId === 'x-scheduler') {
     return `/x/api/scheduler/${kebabCase(component)}/`;
   }
+  // Once mui/material-ui#48209 will be merged we need to remove it
+  if (productId === 'x-chat') {
+    return `/x/api/chat/${kebabCase(component)}/`;
+  }
   if (componentPkg === 'mui-base' || BaseUIReexportedComponents.includes(component)) {
     return `/base-ui/react-${kebabCase(component)}/components-api/#${kebabCase(component)}`;
   }
