diff --git a/lib/common/resolve.js b/lib/common/resolve.js
index 03d4544a15a18c9cfe6b3faeeedc5d30d589bb85..673a3b71b3b45dd34631819d3daa60235036686c 100644
--- a/lib/common/resolve.js
+++ b/lib/common/resolve.js
@@ -240,9 +240,13 @@ function resolve(p, context) {
     try {
         return relative(
             p,
-            context.getPhysicalFilename
-                ? context.getPhysicalFilename()
-                : context.getFilename(),
+            context.physicalFilename != null
+                ? context.physicalFilename
+                : context.filename != null
+                    ? context.filename
+                    : context.getPhysicalFilename
+                        ? context.getPhysicalFilename()
+                        : context.getFilename(),
             context.settings
         );
     } catch (err) {
diff --git a/lib/rules/default-export-match-filename.js b/lib/rules/default-export-match-filename.js
index 765ae31521f85c74e4c3353865be8bb87a0b5b7b..85cb7e5b271251439f07e10c86feb8b378e07596 100644
--- a/lib/rules/default-export-match-filename.js
+++ b/lib/rules/default-export-match-filename.js
@@ -97,7 +97,7 @@ module.exports = {
                     replacePattern = context.options[1]
                         ? new RegExp(context.options[1])
                         : null,
-                    filename = context.getFilename(),
+                    filename = context.filename != null ? context.filename : context.getFilename(),
                     absoluteFilename = path.resolve(filename),
                     parsed = parseFilename(absoluteFilename),
                     shouldIgnore = isIgnoredFilename(filename),
diff --git a/lib/rules/default-import-match-filename.js b/lib/rules/default-import-match-filename.js
index a5d7c770162f47aed2dcc1d29a21178021cfee2d..e234eb3f6881026f167da78e7c984e152cbef6dd 100644
--- a/lib/rules/default-import-match-filename.js
+++ b/lib/rules/default-import-match-filename.js
@@ -84,7 +84,7 @@ function getPackageJsonName(packageJsonPath) {
 }
 
 function getNameFromPackageJsonOrDirname(path, context) {
-    const directoryName = Path.join(context.getFilename(), path, '..');
+    const directoryName = Path.join(context.filename != null ? context.filename : context.getFilename(), path, '..');
     const packageJsonPath = Path.join(directoryName, 'package.json');
     const packageJsonName = getPackageJsonName(packageJsonPath);
     return packageJsonName || Path.basename(directoryName);
