diff --git a/node_modules/karma-mocha/lib/adapter.js b/node_modules/karma-mocha/lib/adapter.js
index 0c8199e..0482a21 100644
--- a/node_modules/karma-mocha/lib/adapter.js
+++ b/node_modules/karma-mocha/lib/adapter.js
@@ -55,10 +55,11 @@ var processAssertionError = function (error_) {
       showDiff: error_.showDiff
     }
 
-    if (error.showDiff) {
-      error.actual = window.Mocha.utils.stringify(error_.actual)
-      error.expected = window.Mocha.utils.stringify(error_.expected)
-    }
+    // See https://github.com/mochajs/mocha/issues/4910#issuecomment-1224341701
+    // if (error.showDiff) {
+    //   error.actual = window.Mocha.utils.stringify(error_.actual)
+    //   error.expected = window.Mocha.utils.stringify(error_.expected)
+    // }
   }
 
   return error
