diff --git a/dist/rasterizeHTML.d.ts b/dist/rasterizeHTML.d.ts
index fa9940c72d1fa59f24d8004116932e7f11b6dd64..89d2a25e83b000dc2a08ad4fe975f12b0d9301df 100644
--- a/dist/rasterizeHTML.d.ts
+++ b/dist/rasterizeHTML.d.ts
@@ -89,6 +89,7 @@ export interface Options {
      * than none. Should be initialized with {}.
      */
     cacheBucket?: {};
+    nonce?: string;
 }
 
 /** Describes a resource that failed to load during drawing. */
